icu4c/source/common/cjmlbe.cpp

```diff
@@ -0,0 +1,457 @@
+        if (U_FAILURE(status)) { return; }
+    }
+    if (w2 != INVALID && w3 != INVALID) {
+        arr[0] = w2;
+        arr[1] = w3;
+        arr[2] = -1;
+        featureList.adoptElement(concatChar(u"BW1:", arr, status), status);
+        if (U_FAILURE(status)) { return; }
+    }
+    if (w3 != INVALID && w4 != INVALID) {
+        arr[0] = w3;
+        arr[1] = w4;
+        arr[2] = -1;
+        featureList.adoptElement(concatChar(u"BW2:", arr, status), status);
+        if (U_FAILURE(status)) { return; }
+    }
+    if (w4 != INVALID && w5 != INVALID) {
+        arr[0] = w4;
+        arr[1] = w5;
+        arr[2] = -1;
+        featureList.adoptElement(concatChar(u"BW3:", arr, status), status);
+        if (U_FAILURE(status)) { return; }
+    }
+    if (w1 != INVALID && w2 != INVALID && w3 != INVALID) {
+        arr[0] = w1;
+        arr[1] = w2;
+        arr[2] = w3;
+        arr[3] = -1;
+        featureList.adoptElement(concatChar(u"TW1:", arr, status), status);
+        if (U_FAILURE(status)) { return; }
+    }
+    if (w2 != INVALID && w3 != INVALID && w4 != INVALID) {
+        arr[0] = w2;
+        arr[1] = w3;
+        arr[2] = w4;
+        arr[3] = -1;
+        featureList.adoptElement(concatChar(u"TW2:", arr, status), status);
+    }
+    if (w3 != INVALID && w4 != INVALID && w5 != INVALID) {
+        arr[0] = w3;
+        arr[1] = w4;
+        arr[2] = w5;
+        arr[3] = -1;
+        featureList.adoptElement(concatChar(u"TW3:", arr, status), status);
+        if (U_FAILURE(status)) { return; }
+    }
+    if (w4 != INVALID && w5 != INVALID && w6 != INVALID) {
+        arr[0] = w4;
+        arr[1] = w5;
+        arr[2] = w6;
+        arr[3] = -1;
+        featureList.adoptElement(concatChar(u"TW4:", arr, status), status);
+    }
+    if (isValid(b1)) {
+        featureList.adoptElement(cloneString(UnicodeString(u"UB1:").append(b1), status), status);
+        if (U_FAILURE(status)) { return; }
+    }
+    if (isValid(b2)) {
+        featureList.adoptElement(cloneString(UnicodeString(u"UB2:").append(b2), status), status);
+        if (U_FAILURE(status)) { return; }
+    }
+    if (isValid(b3)) {
+        featureList.adoptElement(cloneString(UnicodeString(u"UB3:").append(b3), status), status);
+        if (U_FAILURE(status)) { return; }
+    }
+    if (isValid(b4)) {
+        featureList.adoptElement(cloneString(UnicodeString(u"UB4:").append(b4), status), status);
+        if (U_FAILURE(status)) { return; }
+    }
+    if (isValid(b5)) {
+        featureList.adoptElement(cloneString(UnicodeString(u"UB5:").append(b5), status), status);
+        if (U_FAILURE(status)) { return; }
+    }
+    if (isValid(b6)) {
+        featureList.adoptElement(cloneString(UnicodeString(u"UB6:").append(b6), status), status);
+        if (U_FAILURE(status)) { return; }
+    }
+    if (isValid(b2) && isValid(b3)) {
+        featureList.adoptElement(cloneString(UnicodeString(u"BB1:").append(b2).append(b3), status), status);
```

Comment:
these lines are too long and hard to read. Could you wrap the line at most 100 columns?

Reply:
Done